### ISSUE TITLE
Revive not ignoring gRPC-generated files

### ISSUE THREAD

**Issue by peake100** [mgechev/revive#414]
**Describe the bug**
Hi! I have been using revive for a while now and love it! I recently started exploring gRPC and ran into an issue where revive is not ignoring the generated files, even though the setting is turned on.

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```shell
# flags

revive -config revive.toml ./...
```

```toml
# When set to false, ignores files with "GENERATED" header, similar to golint
ignoreGeneratedHeader = true

# Sets the default severity to "warning"
severity = "error"

# Sets the default failure confidence. This means that linting errors
# with less than 0.8 confidence will be ignored.
confidence = 0.8

# Sets the error code for failures with severity "error"
errorCode = 1

# Sets the error code for failures with severity "warning"
warningCode = 1

# Check for commonly mistaken usages of the sync/atomic package
[rule.atomic]

# Warns on bare (a.k.a. naked) returns
[rule.bare-return]

# Blank import should be only in a main or test package, or have a comment justifying it.
[rule.blank-imports]

# Using Boolean literals (true, false) in logic expressions may make the code less
# readable. This rule suggests removing Boolean literals from logic expressions.
[rule.bool-literal-in-expr]

# Explicitly invoking the garbage collector is, except for specific uses in
# benchmarking, very dubious.
[rule.call-to-gc]

# Cognitive complexity is a measure of how hard code is to understand. While cyclomatic
# complexity is good to measure "testeability" of the code, cognitive complexity aims to
# provide a more precise measure of the difficulty of understanding the code. Enforcing
# a maximum complexity per function helps to keep code readable and maintainable.
[rule.cognitive-complexity]
  arguments =[8]

# Methods or fields of struct that have names different only by capitalization could be
# confusing.
[rule.confusing-naming]

# unction or methods that return multiple, no named, values of the same type could
# induce error.
[rule.confusing-results]

# The rule spots logical expressions that evaluate always to the same value.
[rule.constant-logical-expr]

# By convention, context.Context should be the first parameter of a function. This rule
# spots function declarations that do not follow the convention.
[rule.context-as-argument]

# Configuration of the `cyclomatic` rule. Here we specify that
# the rule should fail if it detects code with higher complexity than 10.
[rule.cyclomatic]
    arguments = [10]

# Packages exposing functions that can stop program execution by exiting are hard to
# reuse. This rule looks for program exits in functions other than main() or init().
[rule.deep-exit]

# mporting with . makes the programs much harder to understand because it is unclear
# whether names belong to the current package or to an imported package.
[rule.dot-imports]

# It is possible to unintentionally import the same package twice. This rule looks for
# packages that are imported two or more times.
[rule.duplicated-imports]

# Empty blocks make code less readable and could be a symptom of a bug or unfinished
# refactoring.
[rule.empty-block]

# By convention, for the sake of readability, variables of type error must be named with
# the prefix err.
[rule.error-naming]

# By convention, for better readability, error messages should not be capitalized or end
# with punctuation or a newline.
[rule.error-strings]

# Exported function and methods should have comments. This warns on undocumented
# exported functions and methods.
# [rule.exported]

# It is possible to get a simpler program by replacing errors.New(fmt.Sprintf()) with
# fmt.Errorf(). This rule spots that kind of simplification opportunities.
[rule.errorf]

# Functions returning too many results can be hard to understand/use.
[rule.function-result-limit]
  arguments =[3]

# Typically, functions with names prefixed with Get are supposed to return a value.
[rule.get-return]

# Checking if an error is nil to just after return the error or nil is redundant.
[rule.if-return]

# By convention, for better readability, incrementing an integer variable by 1 is
# recommended to be done using the ++ operator. This rule spots expressions like i += 1
# and i -= 1 and proposes to change them into i++ and i--.
[rule.increment-decrement]

# To improve the readability of code, it is recommended to reduce the indentation as
# much as possible. This rule highlights redundant else-blocks that can be eliminated
# from the code.
[rule.indent-error-flow]

# In GO it is possible to declare identifiers (packages, structs, interfaces,
# parameters, receivers, variables, constants...) that conflict with the name of an
# imported package. This rule spots identifiers that shadow an import.
[rule.import-shadowing]

# Warns in the presence of code lines longer than a configured maximum.
[rule.line-length-limit]
    arguments = [88]

# A method that modifies its receiver value can have undesired behavior.
# The modification can be also the root of a bug because the actual value receiver could
# be a copy of that used at the calling site. This rule warns when a method modifies its
# receiver.
[rule.modifies-value-receiver]

# This rule suggests a shorter way of writing ranges that do not use the second value.
[rule.range]

# Range variables in a loop are reused at each iteration; therefore a goroutine created
# in a loop will point to the range variable with from the upper scope. This way, the
# goroutine could use the variable with an undesired value. This rule warns when a range
# value (or index) is used inside a closure
[rule.range-val-in-closure]

# By convention, receiver names in a method should reflect their identity. For example,
# if the receiver is of type Parts, p is an adequate name for it. Contrary to other
# languages, it is not idiomatic to name receivers as this or self.
[rule.receiver-naming]

# Constant names like false, true, nil, function names like append, make, and basic type
# names like bool, and byte are not reserved words of the language; therefore the can be
# redefined. Even if possible, redefining these built in names can lead to bugs very
# difficult to detect.
[rule.redefines-builtin-id]

# Struct tags are not checked at compile time. This rule, checks and warns if it finds
# errors in common struct tags types like: asn1, default, json, protobuf, xml, yaml.
[rule.struct-tag]

# To improve the readability of code, it is recommended to reduce the indentation as
# much as possible. This rule highlights redundant else-blocks that can be eliminated
# from the code.
[rule.superfluous-else]

# This rule warns when variable or package naming conventions are not followed.
[rule.var-naming]
    arguments = [["HTTP", "ID", "JSON", "API"],[]]

# This rule proposes simplifications of variable declarations.
[rule.var-declaration]

# This rule warns when errors returned by a function are not explicitly handled on the
# caller side.
[rule.unhandled-error]

# This rule suggests to remove redundant statements like a break at the end of a case
# block, for improving the code's readability.
[rule.unnecessary-stmt]

# Function parameters that are passed by value, are in fact a copy of the original
# argument. Passing a copy of a sync.WaitGroup is usually not what the developer wants
# to do. This rule warns when a sync.WaitGroup expected as a by-value parameter in a
# function or method.
[rule.waitgroup-by-value]

```

**Expected behavior**
The generated files in my ./proto folder should be ignored, they start with:

```
// Code generated by protoc-gen-go. DO NOT EDIT.
// versions:
// 	protoc-gen-go v1.22.0
// 	protoc        v3.11.4
// source: proto/service.proto
```

**Logs**
```
proto/service.pb.gw.go:9:9: The name 'proto' shadows an import name
proto/service.pb.gw.go:34:6: don't use underscores in Go names; func request_StalkForecaster_ForecastPrices_0 should be requestStalkForecasterForecastPrices0
proto/service.pb.gw.go:51:6: don't use underscores in Go names; func local_request_StalkForecaster_ForecastPrices_0 should be localRequestStalkForecasterForecastPrices0
proto/service.pb.gw.go:158:2: don't use underscores in Go names; var pattern_StalkForecaster_ForecastPrices_0 should be patternStalkForecasterForecastPrices0
proto/service.pb.gw.go:162:2: don't use underscores in Go names; var forward_StalkForecaster_ForecastPrices_0 should be forwardStalkForecasterForecastPrices0
proto/service.pb.go:12:2: a blank import should be only in a main or test package, or have a comment justifying it
proto/service.pb.go:7:9: The name 'proto' shadows an import name
proto/service.pb.go:37:2: don't use underscores in Go names; const PricePatterns_FLUCTUATING should be PricePatternsFLUCTUATING
proto/service.pb.go:38:2: don't use underscores in Go names; const PricePatterns_BIGSPIKE should be PricePatternsBIGSPIKE
proto/service.pb.go:39:2: don't use underscores in Go names; const PricePatterns_DECREASING should be PricePatternsDECREASING
proto/service.pb.go:40:2: don't use underscores in Go names; const PricePatterns_SMALLSPIKE should be PricePatternsSMALLSPIKE
proto/service.pb.go:41:2: don't use underscores in Go names; const PricePatterns_UNKNOWN should be PricePatternsUNKNOWN
proto/service.pb.go:46:2: don't use underscores in Go names; var PricePatterns_name should be PricePatternsName
proto/service.pb.go:53:2: don't use underscores in Go names; var PricePatterns_value should be PricePatternsValue
proto/service.pb.go:744:5: don't use underscores in Go names; var File_proto_service_proto should be FileProtoServiceProto
proto/service.pb.go:746:5: don't use underscores in Go names; var file_proto_service_proto_rawDesc should be fileProtoServiceProtoRawDesc
proto/service.pb.go:850:2: don't use underscores in Go names; var file_proto_service_proto_rawDescOnce should be fileProtoServiceProtoRawDescOnce
proto/service.pb.go:851:2: don't use underscores in Go names; var file_proto_service_proto_rawDescData should be fileProtoServiceProtoRawDescData
proto/service.pb.go:854:6: don't use underscores in Go names; func file_proto_service_proto_rawDescGZIP should be fileProtoServiceProtoRawDescGZIP
proto/service.pb.go:861:5: don't use underscores in Go names; var file_proto_service_proto_enumTypes should be fileProtoServiceProtoEnumTypes
proto/service.pb.go:862:5: don't use underscores in Go names; var file_proto_service_proto_msgTypes should be fileProtoServiceProtoMsgTypes
proto/service.pb.go:863:5: don't use underscores in Go names; var file_proto_service_proto_goTypes should be fileProtoServiceProtoGoTypes
proto/service.pb.go:875:5: don't use underscores in Go names; var file_proto_service_proto_depIdxs should be fileProtoServiceProtoDepIdxs
proto/service.pb.go:900:6: don't use underscores in Go names; func file_proto_service_proto_init should be fileProtoServiceProtoInit
proto/service.pb.go:1086:6: don't use underscores in Go names; func _StalkForecaster_ForecastPrices_Handler should be _StalkForecasterForecastPricesHandler
proto/service.pb.go:1104:5: don't use underscores in Go names; var _StalkForecaster_serviceDesc should be _StalkForecasterServiceDesc
proto/service.pb.gw.go:34: line is 223 characters, out of limit 88
proto/service.pb.gw.go:42: line is 96 characters, out of limit 88
proto/service.pb.gw.go:46: line is 121 characters, out of limit 88
proto/service.pb.gw.go:51: line is 229 characters, out of limit 88
proto/service.pb.gw.go:59: line is 96 characters, out of limit 88
proto/service.pb.gw.go:68: line is 105 characters, out of limit 88
proto/service.pb.gw.go:70: line is 90 characters, out of limit 88
proto/service.pb.gw.go:71: line is 123 characters, out of limit 88
proto/service.pb.gw.go:73: line is 143 characters, out of limit 88
proto/service.pb.gw.go:82: line is 120 characters, out of limit 88
proto/service.pb.gw.go:89: line is 127 characters, out of limit 88
proto/service.pb.gw.go:96: line is 91 characters, out of limit 88
proto/service.pb.gw.go:98: line is 146 characters, out of limit 88
proto/service.pb.gw.go:121: line is 99 characters, out of limit 88
proto/service.pb.gw.go:123: line is 110 characters, out of limit 88
proto/service.pb.gw.go:124: line is 89 characters, out of limit 88
proto/service.pb.gw.go:127: line is 95 characters, out of limit 88
proto/service.pb.gw.go:128: line is 121 characters, out of limit 88
proto/service.pb.gw.go:129: line is 115 characters, out of limit 88
proto/service.pb.gw.go:130: line is 108 characters, out of limit 88
proto/service.pb.gw.go:132: line is 123 characters, out of limit 88
proto/service.pb.gw.go:134: line is 143 characters, out of limit 88
proto/service.pb.gw.go:143: line is 114 characters, out of limit 88
proto/service.pb.gw.go:150: line is 127 characters, out of limit 88
proto/service.pb.gw.go:158: line is 162 characters, out of limit 88
proto/service.pb.gw.go:98:1: function RegisterStalkForecasterHandlerFromEndpoint has cognitive complexity 10 (> max enabled 8)
proto/service.pb.go:900:1: function file_proto_service_proto_init has cyclomatic complexity 39
proto/service.pb.go:96: line is 127 characters, out of limit 88
proto/service.pb.go:98: line is 168 characters, out of limit 88
proto/service.pb.go:103: line is 94 characters, out of limit 88
proto/service.pb.go:169: line is 103 characters, out of limit 88
proto/service.pb.go:237: line is 124 characters, out of limit 88
proto/service.pb.go:239: line is 124 characters, out of limit 88
proto/service.pb.go:378: line is 93 characters, out of limit 88
proto/service.pb.go:380: line is 135 characters, out of limit 88
proto/service.pb.go:452: line is 131 characters, out of limit 88
proto/service.pb.go:453: line is 105 characters, out of limit 88
proto/service.pb.go:454: line is 138 characters, out of limit 88
proto/service.pb.go:455: line is 101 characters, out of limit 88
proto/service.pb.go:456: line is 141 characters, out of limit 88
proto/service.pb.go:542: line is 106 characters, out of limit 88
proto/service.pb.go:620: line is 89 characters, out of limit 88
proto/service.pb.go:686: line is 140 characters, out of limit 88
proto/service.pb.go:687: line is 105 characters, out of limit 88
proto/service.pb.go:688: line is 109 characters, out of limit 88
proto/service.pb.go:747: line is 99 characters, out of limit 88
proto/service.pb.go:748: line is 99 characters, out of limit 88
proto/service.pb.go:749: line is 99 characters, out of limit 88
proto/service.pb.go:750: line is 99 characters, out of limit 88
proto/service.pb.go:751: line is 99 characters, out of limit 88
proto/service.pb.go:752: line is 99 characters, out of limit 88
proto/service.pb.go:753: line is 99 characters, out of limit 88
proto/service.pb.go:754: line is 99 characters, out of limit 88
proto/service.pb.go:755: line is 99 characters, out of limit 88
proto/service.pb.go:756: line is 99 characters, out of limit 88
proto/service.pb.go:757: line is 99 characters, out of limit 88
proto/service.pb.go:758: line is 99 characters, out of limit 88
proto/service.pb.go:759: line is 99 characters, out of limit 88
proto/service.pb.go:760: line is 99 characters, out of limit 88
proto/service.pb.go:761: line is 99 characters, out of limit 88
proto/service.pb.go:762: line is 99 characters, out of limit 88
proto/service.pb.go:763: line is 99 characters, out of limit 88
proto/service.pb.go:764: line is 99 characters, out of limit 88
proto/service.pb.go:765: line is 99 characters, out of limit 88
proto/service.pb.go:766: line is 99 characters, out of limit 88
proto/service.pb.go:767: line is 99 characters, out of limit 88
proto/service.pb.go:768: line is 99 characters, out of limit 88
proto/service.pb.go:769: line is 99 characters, out of limit 88
proto/service.pb.go:770: line is 99 characters, out of limit 88
proto/service.pb.go:771: line is 99 characters, out of limit 88
proto/service.pb.go:772: line is 99 characters, out of limit 88
proto/service.pb.go:773: line is 99 characters, out of limit 88
proto/service.pb.go:774: line is 99 characters, out of limit 88
proto/service.pb.go:775: line is 99 characters, out of limit 88
proto/service.pb.go:776: line is 99 characters, out of limit 88
proto/service.pb.go:777: line is 99 characters, out of limit 88
proto/service.pb.go:778: line is 99 characters, out of limit 88
proto/service.pb.go:779: line is 99 characters, out of limit 88
proto/service.pb.go:780: line is 99 characters, out of limit 88
proto/service.pb.go:781: line is 99 characters, out of limit 88
proto/service.pb.go:782: line is 99 characters, out of limit 88
proto/service.pb.go:783: line is 99 characters, out of limit 88
proto/service.pb.go:784: line is 99 characters, out of limit 88
proto/service.pb.go:785: line is 99 characters, out of limit 88
proto/service.pb.go:786: line is 99 characters, out of limit 88
proto/service.pb.go:787: line is 99 characters, out of limit 88
proto/service.pb.go:788: line is 99 characters, out of limit 88
proto/service.pb.go:789: line is 99 characters, out of limit 88
proto/service.pb.go:790: line is 99 characters, out of limit 88
proto/service.pb.go:791: line is 99 characters, out of limit 88
proto/service.pb.go:792: line is 99 characters, out of limit 88
proto/service.pb.go:793: line is 99 characters, out of limit 88
proto/service.pb.go:794: line is 99 characters, out of limit 88
proto/service.pb.go:795: line is 99 characters, out of limit 88
proto/service.pb.go:796: line is 99 characters, out of limit 88
proto/service.pb.go:797: line is 99 characters, out of limit 88
proto/service.pb.go:798: line is 99 characters, out of limit 88
proto/service.pb.go:799: line is 99 characters, out of limit 88
proto/service.pb.go:800: line is 99 characters, out of limit 88
proto/service.pb.go:801: line is 99 characters, out of limit 88
proto/service.pb.go:802: line is 99 characters, out of limit 88
proto/service.pb.go:803: line is 99 characters, out of limit 88
proto/service.pb.go:804: line is 99 characters, out of limit 88
proto/service.pb.go:805: line is 99 characters, out of limit 88
proto/service.pb.go:806: line is 99 characters, out of limit 88
proto/service.pb.go:807: line is 99 characters, out of limit 88
proto/service.pb.go:808: line is 99 characters, out of limit 88
proto/service.pb.go:809: line is 99 characters, out of limit 88
proto/service.pb.go:810: line is 99 characters, out of limit 88
proto/service.pb.go:811: line is 99 characters, out of limit 88
proto/service.pb.go:812: line is 99 characters, out of limit 88
proto/service.pb.go:813: line is 99 characters, out of limit 88
proto/service.pb.go:814: line is 99 characters, out of limit 88
proto/service.pb.go:815: line is 99 characters, out of limit 88
proto/service.pb.go:816: line is 99 characters, out of limit 88
proto/service.pb.go:817: line is 99 characters, out of limit 88
proto/service.pb.go:818: line is 99 characters, out of limit 88
proto/service.pb.go:819: line is 99 characters, out of limit 88
proto/service.pb.go:820: line is 99 characters, out of limit 88
proto/service.pb.go:821: line is 99 characters, out of limit 88
proto/service.pb.go:822: line is 99 characters, out of limit 88
proto/service.pb.go:823: line is 99 characters, out of limit 88
proto/service.pb.go:824: line is 99 characters, out of limit 88
proto/service.pb.go:825: line is 99 characters, out of limit 88
proto/service.pb.go:826: line is 99 characters, out of limit 88
proto/service.pb.go:827: line is 99 characters, out of limit 88
proto/service.pb.go:828: line is 99 characters, out of limit 88
proto/service.pb.go:829: line is 99 characters, out of limit 88
proto/service.pb.go:830: line is 99 characters, out of limit 88
proto/service.pb.go:831: line is 99 characters, out of limit 88
proto/service.pb.go:832: line is 99 characters, out of limit 88
proto/service.pb.go:833: line is 99 characters, out of limit 88
proto/service.pb.go:834: line is 99 characters, out of limit 88
proto/service.pb.go:835: line is 99 characters, out of limit 88
proto/service.pb.go:836: line is 99 characters, out of limit 88
proto/service.pb.go:837: line is 99 characters, out of limit 88
proto/service.pb.go:838: line is 99 characters, out of limit 88
proto/service.pb.go:839: line is 99 characters, out of limit 88
proto/service.pb.go:840: line is 99 characters, out of limit 88
proto/service.pb.go:841: line is 99 characters, out of limit 88
proto/service.pb.go:842: line is 99 characters, out of limit 88
proto/service.pb.go:843: line is 99 characters, out of limit 88
proto/service.pb.go:844: line is 99 characters, out of limit 88
proto/service.pb.go:845: line is 99 characters, out of limit 88
proto/service.pb.go:856: line is 109 characters, out of limit 88
proto/service.pb.go:905: line is 96 characters, out of limit 88
proto/service.pb.go:917: line is 96 characters, out of limit 88
proto/service.pb.go:929: line is 96 characters, out of limit 88
proto/service.pb.go:941: line is 96 characters, out of limit 88
proto/service.pb.go:953: line is 96 characters, out of limit 88
proto/service.pb.go:965: line is 96 characters, out of limit 88
proto/service.pb.go:977: line is 96 characters, out of limit 88
proto/service.pb.go:989: line is 96 characters, out of limit 88
proto/service.pb.go:1001: line is 96 characters, out of limit 88
proto/service.pb.go:1045: line is 145 characters, out of limit 88
proto/service.pb.go:1048: line is 95 characters, out of limit 88
proto/service.pb.go:1059: line is 125 characters, out of limit 88
proto/service.pb.go:1074: line is 97 characters, out of limit 88
proto/service.pb.go:1078: line is 104 characters, out of limit 88
proto/service.pb.go:1079: line is 91 characters, out of limit 88
proto/service.pb.go:1086: line is 175 characters, out of limit 88
proto/service.pb.go:900:1: function file_proto_service_proto_init has cognitive complexity 38 (> max enabled 8)
proto/service.pb.go:1086:1: context.Context should be the first parameter of a function
```

**Desktop (please complete the following information):**
 - OS: Mac OSX Mojave 10.14.1
 - Version of Go: go1.14 darwin/amd64

**Additional context**
Using exclude flags in the CLI command is not a good solution for me. Our CI/CD pipeline uses a common template for go projects, so project-specific ignores like this would be better handled at the revive config level.

**Comment by chavacava** [comment]
Hi @peake100, thanks for filling the issue.
Did you try setting `ignoreGeneratedHeader` to `false`?

```toml
# When set to false, ignores files with "GENERATED" header, similar to golint
ignoreGeneratedHeader = false
```

**Comment by peake100** [comment]
Thanks, that fixed it. I think my brain automatically interpreted that option as ignoring files with generated headers, not ignoring the headers and thereby processing the files.

Should have read the docs more carefully. Thanks for your quick response!
